worlds/messenger/SubClasses.py

```diff
@@ -0,0 +1,61 @@
+from typing import Set, TYPE_CHECKING
+
+from BaseClasses import Region, Location, Item, ItemClassification, Entrance
+from .Constants import SEALS, NOTES, PROG_ITEMS, PHOBEKINS, USEFUL_ITEMS
+from .Options import Goal
+from .Regions import REGIONS
+
+if TYPE_CHECKING:
+    from . import MessengerWorld
+else:
+    MessengerWorld = object
+
+
+class MessengerRegion(Region):
+    def __init__(self, name: str, world: MessengerWorld):
+        super().__init__(name, world.player, world.multiworld)
+        self.locations_lookup = self.multiworld.worlds[self.player].location_name_to_id
+        self.add_locations()
+        world.multiworld.regions.append(self)
+
+    def add_locations(self) -> None:
+        for loc in REGIONS[self.name]:
+            self.locations.append(MessengerLocation(loc, self))
+        if self.name == "The Shop" and self.multiworld.goal[self.player] > Goal.option_open_music_box:
+            self.locations.append(MessengerLocation("Shop Chest", self))
+        if self.multiworld.shuffle_seals[self.player]:
+            # putting some dumb special case for searing crags and ToT so i can split them into 2 regions
+            seal_locs = [loc for loc in SEALS
+                         if loc.startswith(self.name.split(" ")[0]) and self.name not in {"Searing Crags", "Tower HQ"}]
+            for seal_loc in seal_locs:
+                self.locations.append(MessengerLocation(seal_loc, self))
+
+    def add_exits(self, exits: Set[str]) -> None:
+        for exit in exits:
+            ret = Entrance(self.player, f"{self.name} -> {exit}", self)
+            self.exits.append(ret)
+            ret.connect(self.multiworld.get_region(exit, self.player))
+
+
+class MessengerLocation(Location):
+    game = "The Messenger"
+
+    def __init__(self, name: str, parent: MessengerRegion):
+        loc_id = parent.locations_lookup[name] if name in parent.locations_lookup else None
+        super().__init__(parent.player, name, loc_id, parent)
+        if loc_id is None:
+            self.place_locked_item(parent.multiworld.worlds[self.player].create_item(name))
+
+
+class MessengerItem(Item):
+    game = "The Messenger"
+
+    def __init__(self, name: str, player: int, item_id: int = None):
+        if name in {*NOTES, *PROG_ITEMS, *PHOBEKINS} or item_id is None:
+            item_class = ItemClassification.progression
```

Comment:
Possibly. It requires all of the notes for a single location and all phobekins for a different single location, but with seals shuffled phobekins actually block 4 locations instead of 1. This is something I've thought about but I want to see how it works in practice before doing.